alternator/executor.cc

```diff
@@ -1509,7 +1525,7 @@ static future<std::unique_ptr<rjson::value>> get_previous_item(
     command->allow_limit = db::allow_per_partition_rate_limit::yes;
     auto cl = db::consistency_level::LOCAL_QUORUM;
 
-    return proxy.query(schema, command, to_partition_ranges(*schema, pk), cl, service::storage_proxy::coordinator_query_options(executor::default_timeout(), std::move(permit), client_state)).then(
+    return proxy.query(schema, command, to_partition_ranges(*schema, pk), cl, service::storage_proxy::coordinator_query_options(executor::default_timeout(executor::default_getitem_timeout), std::move(permit), client_state)).then(
```

Comment:
Oh, I see you use the getitem_timeout also for the read-before-write of **write** operations. That wasn't obvious from the name of the timeout. Maybe better call it "default_read_item_timeout", to avoid giving the impression that it's only for GetItem. I kept the word "item" to emphasize it's for reading a single item (and not a batch or scan), but I'm not sure we need this emphesis or distinction.

Reply:
Also I see below that for LWT, if I understand correctly, you have just one (putitem_lwt) timeout for the entire read-modify-write operation?